gravitee-am-test/specs/gateway/account-link.jest.spec.ts

```diff
@@ -0,0 +1,330 @@
+  domainAccLinking = masterDomainStarted.domain;
+  domainAccLinkingConfiguration = masterDomainStarted.oidcConfig;
+
+  //Enable bruteforce
+  await patchDomain(domainAccLinking.id, accessToken, {
+    accountSettings: {
+      maxLoginAttempts: 3,
+      loginAttemptsDetectionEnabled: true,
+      accountBlockedDuration: 180,
+      loginAttemptsResetTime: 120,
+    },
+  });
+
+  //Create oidc provider
+  const idpConfig = {
+    clientId: 'oidc',
+    clientSecret: 'oidc',
+    clientAuthenticationMethod: 'client_secret_basic',
+    wellKnownUri: 'http://localhost:8092/oidc-domain/oidc/.well-known/openid-configuration',
+    userAuthorizationUri: 'http://localhost:8092/oidc-domain/oauth/authorize',
+    accessTokenUri: 'http://localhost:8092/oidc-domain/oauth/token',
+    userProfileUri: 'http://localhost:8092/oidc-domain/oidc/userinfo',
+    logoutUri: 'http://localhost:8092/oidc-domain/logout',
+    responseType: 'code',
+    responseMode: 'default',
+    encodeRedirectUri: false,
+    useIdTokenForUserInfo: false,
+    signature: 'RSA_RS256',
+    publicKeyResolver: 'GIVEN_KEY',
+    connectTimeout: 10000,
+    idleTimeout: 10000,
+    maxPoolSize: 200,
+    storeOriginalTokens: false,
+  };
+  let openIdIdp = {
+    configuration: `${JSON.stringify(idpConfig)}`,
+    name: 'oidc',
+    type: 'oauth2-generic-am-idp',
+    external: true,
+  };
+  const oidcIdp = await createIdp(domainAccLinking.id, accessToken, openIdIdp);
+
+  // Create an application with oidc provider
+  applicationAccLinking = await createApplication(domainAccLinking.id, accessToken, {
+    name: 'app',
+    type: 'WEB',
+    clientId: 'app',
+    clientSecret: 'app',
+    redirectUris: ['https://test.com'],
+  }).then((app) =>
+    updateApplication(
+      domainAccLinking.id,
+      accessToken,
+      {
+        identityProviders: [{ identity: oidcIdp.id, priority: -1 }],
+        settings: {
+          login: {
+            inherited: false,
+            hideForm: true,
+          },
+        },
+      },
+      app.id,
+    ),
+  );
+
+  // Create an application with an internal provider
+  const idpSetAccountLinkingDomain = await getAllIdps(domainAccLinking.id, accessToken);
+  const localIdp = idpSetAccountLinkingDomain.filter((idp) => idp.type === 'mongo-am-idp' || 'jdbc-am-idp');
+  localIdpApp = await createApplication(domainAccLinking.id, accessToken, {
+    name: 'app-local',
+    type: 'WEB',
+    clientId: 'app-local',
+    clientSecret: 'app-local',
+    redirectUris: ['https://test.com'],
+  }).then((app) =>
+    updateApplication(
+      domainAccLinking.id,
+      accessToken,
+      {
+        identityProviders: [{ identity: localIdp.values().next().value.id, priority: -1 }],
+      },
+      app.id,
+    ),
+  );
+  domainUser = getUser('domain-user');
+  // Create the same user
+  await createUser(domainAccLinking.id, accessToken, domainUser);
+  await delay(6000);
```

Comment:
```suggestion
  await waitForDomainSync();
```

Reply:
waitForDomainSync was initially 6s but we had troubles that's why I set it to 10.
I would prefer to use the domainSync domain even if it is waiting 4s more at least we clearly highlight why we are waiting and if we succeed to provide internal API to have the sync state, we have only method to update